server.go

```diff
@@ -303,8 +305,11 @@ func (srv *Server) init() {
 	if srv.MsgAcceptFunc == nil {
 		srv.MsgAcceptFunc = defaultMsgAcceptFunc
 	}
+	if srv.SessionUDPFactory == nil {
+		srv.SessionUDPFactory = defaultSessionUDPFactory
+	}
 
-	srv.udpPool.New = makeUDPBuffer(srv.UDPSize)
+	srv.udpPool.New = srv.makeSessionUDP
```

Comment:
This causes a GC loop and prevents `srv` from ever being collected.

Reply:
Would changing the above to this remove the GC loop?
```
	udpFactory := srv.SessionUDPFactory
	udpSize := srv.UDPSize
	srv.udpPool.New = func () interface{} {
		return udpFactory.CreateSessionUDP(udpSize)
	}
```